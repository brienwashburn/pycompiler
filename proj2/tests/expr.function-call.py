foo()
foo(a)
foo(a,)
foo(a=10)
foo(a,b)
foo(a,b=10)
foo(a=10,b=10)
foo(a,b,c,*d)
foo(a,b,c,*d,e=10)
foo(a,b,c=10,*d,**kw)
foo(a,b,c=10,*d,e=10,**kw)
foo(*d,c=10,d=20,**kw)
foo(**kw)

foo(a)(b)(c,d)(a=10,**kw)
