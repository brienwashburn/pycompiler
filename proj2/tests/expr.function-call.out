(Module
 (Expr (Call (func (Name foo)) (args) (keywords) (starargs #f) (kwargs #f)))
 (Expr
  (Call
   (func (Name foo))
   (args (Name a))
   (keywords)
   (starargs #f)
   (kwargs #f)))
 (Expr
  (Call
   (func (Name foo))
   (args (Name a))
   (keywords)
   (starargs #f)
   (kwargs #f)))
 (Expr
  (Call
   (func (Name foo))
   (args)
   (keywords (a (Num 10)))
   (starargs #f)
   (kwargs #f)))
 (Expr
  (Call
   (func (Name foo))
   (args (Name a) (Name b))
   (keywords)
   (starargs #f)
   (kwargs #f)))
 (Expr
  (Call
   (func (Name foo))
   (args (Name a))
   (keywords (b (Num 10)))
   (starargs #f)
   (kwargs #f)))
 (Expr
  (Call
   (func (Name foo))
   (args)
   (keywords (a (Num 10)) (b (Num 10)))
   (starargs #f)
   (kwargs #f)))
 (Expr
  (Call
   (func (Name foo))
   (args (Name a) (Name b) (Name c))
   (keywords)
   (starargs (Name d))
   (kwargs #f)))
 (Expr
  (Call
   (func (Name foo))
   (args (Name a) (Name b) (Name c))
   (keywords (e (Num 10)))
   (starargs (Name d))
   (kwargs #f)))
 (Expr
  (Call
   (func (Name foo))
   (args (Name a) (Name b))
   (keywords (c (Num 10)))
   (starargs (Name d))
   (kwargs (Name kw))))
 (Expr
  (Call
   (func (Name foo))
   (args (Name a) (Name b))
   (keywords (c (Num 10)) (e (Num 10)))
   (starargs (Name d))
   (kwargs (Name kw))))
 (Expr
  (Call
   (func (Name foo))
   (args)
   (keywords (c (Num 10)) (d (Num 20)))
   (starargs (Name d))
   (kwargs (Name kw))))
 (Expr
  (Call (func (Name foo)) (args) (keywords) (starargs #f) (kwargs (Name kw))))
 (Expr
  (Call
   (func
    (Call
     (func
      (Call
       (func
        (Call
         (func (Name foo))
         (args (Name a))
         (keywords)
         (starargs #f)
         (kwargs #f)))
       (args (Name b))
       (keywords)
       (starargs #f)
       (kwargs #f)))
     (args (Name c) (Name d))
     (keywords)
     (starargs #f)
     (kwargs #f)))
   (args)
   (keywords (a (Num 10)))
   (starargs #f)
   (kwargs (Name kw)))))
